template/lib.typ

```diff
@@ -227,27 +227,151 @@ text)
+
+        // Right headers for the main content
+        if locate(loc).page() <= here().page() and title.level == 2 {
+          title_right = strong([#title_numbering ]+ title.body + h(1.4em) + [#numbering(numeration,page)]);
+        }
+
+        // Skip the first header of a section
+        if(title.level == 1 and locate(loc).page() == here().page()){
+          title_left = none;
+        }
+    }
+
+    // Header placement
+    if title_left != none{
+    if calc.odd(page){
+
+    set text(1em)  
+    place(top + right, dx: + 2em, dy: 3.5em,[#title_right] )
+    line(length: 100%, stroke: (paint: gray, thickness: 1pt), start:(0pt,-10pt), end:(40em + 25pt,-10pt))
+
+    } else {
+   
+    set text(1em)  
+    place(top + left, dx: -2em, dy: 3.5em,[#title_left] )
+    line(length: 100%, stroke: (paint: gray, thickness: 1pt), start:(-25pt,-10pt), end:(40em,-10pt))
+  
+    }
+    } else {
+
+      // If there is no header, place the page numbering in the middle
+      place(top + center , dx: -.025em, dy: 72.028em,[#numbering(numeration,page)])
+    }
+   }
+
+  } else {
+
+       context {
+    // Header for the others type of content    
+    let titles = query(selector(heading.where(level: 1)))
+
+    let title_left
+    let num_left
+    let title_right
+    let num_right
+
+    let page = counter(page).get().first() 
+
+    for title in titles {
+    let loc = title.location() 
+    let title_numbering = if title.numbering == none {
+          none
+        } else {
+          context numbering("1.", ..counter(heading).at(loc))
+        } 
+
+        // Prepare both left and right headers parts
+        if locate(loc).page() <= here().page()  {
+          num_left = strong([#numbering(numeration,page)]);
+          title_left = strong(title.body);
+          num_right = strong([#numbering(numeration,page)]);
+          title_right = strong(title.body);
+        }
+
+        // Skip the first header of a section
+        if(locate(loc).page() == here().page()){
+          title_left = none;
+        }
+    }
+
+    // Header placement
+    if title_left != none{
+    if calc.odd(page){
+
+    set text(1em)  
+    place(top + right, dx: + 2em, dy: 3.5em,[#num_right])
+    place(top + center, dx: 0em, dy: 3.5em,[#title_right])
+    line(length: 100%, stroke: (paint: gray, thickness: 1pt), start:(0pt,-10pt), end:(40em + 25pt,-10pt))
+
+    } else {
+   
+    set text(1em)  
+    place(top + left, dx: -2em, dy: 3.5em,[#num_left])
+    place(top + center, dx: + 0em, dy: 3.5em,[#title_left] )
+    line(length: 100%, stroke: (paint: gray, thickness: 1pt), start:(-25pt,-10pt), end:(40em,-10pt))
+  
+    }
+    } else {
+      // If there is no header, place the page numbering in the middle
+      place(top + center , dx: -.025em, dy: 72.028em,[#numbering(numeration,page)])
+    }
+   }
+   }
+  }
 
 #let pre-content(body) = {
-  set page(numbering: "I")
+  set page(numbering:"I", header: pretty-header(numeration: "I", main: false),footer:[])
```

Comment:
Do you need to specify `footer` as well?

Reply:
I mean, shouldn't we be showing the page number if we're using it for the index?

Reply:
Well we are showing it but now it is shown on the top left/right part of the page (unless there is a main heading on the page, then the page number is shown on the bottom middle)

Reply:
Ah I see. Then `footer: none` or this makes sense.